### PR TITLE
[aznamespaces] Fixing a brittle test that depended on the server-side error message

### PR DESCRIPTION
**Pull request by richardpark-msft**

### CODE REVIEW

File: sdk/messaging/eventgrid/aznamespaces/client_test.go [hunk 1]
```diff
@@ -312,11 +312,17 @@ func TestSimpleErrors(t *testing.T) {
 	_, err := c.PublishCloudEvents(context.Background(), c.TestVars.Topic, []messaging.CloudEvent{
 		{},
 	}, nil)
-	var respErr *azcore.ResponseError
 
+	// this'll lead to an error message like this:
+	// "error": {
+	// 	"code":"BadRequest",
```
Comment (jhendrixMSFT) [inline sdk/messaging/eventgrid/aznamespaces/client_test.go:318]: Is there a stable error code we can check for?

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/messaging/eventgrid/aznamespaces/client_test.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/messaging/eventgrid/aznamespaces/client_test.go:318]: But error code?

Comment (richardpark-msft) [reply sdk/messaging/eventgrid/aznamespaces/client_test.go:318]: Added (and in another spot as well)

**Pull request merged**